include/swift/SIL/SILBuilder.h

```diff
@@ -712,6 +712,23 @@ class SILBuilder {
                       BeginBorrowInst(getSILDebugLocation(Loc), LV));
   }
 
+  /// If \p v is compatible with guaranteed, return an empty
+  /// SILValue. Otherwise, borrow \p v and return the borrowed value. If
+  /// ownership is disabled in this function, then the original value is
+  /// returned.
+  ///
+  /// This allows for recursive code that deals with borrows to only borrow the
+  /// first time the function is called. It also by returning v when ownership
+  /// is disabled makes it easy to write code that works on ossa and on non-ossa
+  /// SIL.
```

Comment:
The PR / commit message has great documentation that I think should be in the code, easier to grok than just reading this comment, I would incorporate some of that here.